### PR TITLE
Feature/abstract module and client

### PR DESCRIPTION
**Pull request by ry3s**
* Use `RuntimeModuleBase` as a base class for `RpcRuntimeModule` (old name: `RuntimeClient`)
* Add `get_input_details` and `get_output_details` to RPC 
* Change the return value of `get_input_details` and `get_output_details` 
  * `List[ { name:string, dtype: string, shape: List[int] } ]`
* Update test files related to above changes

### CODE REVIEW

File: python/arachne_runtime/__init__.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-from typing import Optional
+from typing import Dict, Optional
 
 import yaml
 from packaging.version import Version
 
-from .module import RuntimeModuleBase, RuntimeModuleFactory
-from .utils.version_utils import (
+from arachne_runtime.module import RuntimeModuleBase, RuntimeModuleFactory
+from arachne_runtime.rpc.client.client import RpcRuntimeModule
```
Comment (tk1012) [inline python/arachne_runtime/__init__.py]: Unused import

File: python/arachne_runtime/__init__.py [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
+from arachne_runtime.rpc.server import create_channel
```
Comment (tk1012) [inline python/arachne_runtime/__init__.py]: Unused import

**Review by tk1012 — CHANGES_REQUESTED** [review]

File: python/arachne_runtime/__init__.py (hunk 2 above)
Comment (ry3s) [reply python/arachne_runtime/__init__.py]: Deleted.

File: python/arachne_runtime/__init__.py (hunk 1 above)
Comment (ry3s) [reply python/arachne_runtime/__init__.py]: `import arachne_runtime.rpc` is needed to register `RpcRuntimeModule` to the factory

**Review by tk1012 — APPROVED** [review]

**Pull request merged**